### PR TITLE
UIEvents: Test the textInput event

### PR DESCRIPTION
**Pull request by zcorpan**
See https://github.com/w3c/uievents/pull/362

### CODE REVIEW

File: uievents/textInput/smiley-manual.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<script src="/resources/testharnessreport.js"></script>
+<style>
+[contenteditable] { border: thin inset silver; }
+</style>
+<p>Type "🙂" into each text field below.</p>
+<input class=test-el>
+<textarea class=test-el></textarea>
+<div contenteditable=true class=test-el></div>
+<script>
+setup({ explicit_timeout: true });
```
Comment (masayuki-nakano) [inline uievents/textInput/smiley-manual.html:14]: Oh, adding a manual test, unfortunately...

**Review by masayuki-nakano — CHANGES_REQUESTED** [review]

**Comment by masayuki-nakano** [comment]
And could you apply this change?
https://phabricator.services.mozilla.com/differential/changeset/?ref=7325979

File: uievents/textInput/support/common.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    rv += ` contenteditable="${el.getAttribute('contenteditable')}"`;
+  }
+  if (el.hasAttribute('type')) {
+    rv += ` type="${el.getAttribute('type')}"`;
+  }
+  rv += `>`;
+  return rv;
+}
+
+function setSelection(el) {
```
[the remark this answers is not in the document]
Comment (zcorpan) [reply uievents/textInput/support/common.js]: Oops, I meant to do that. Fixed.

File: uievents/textInput/smiley-manual.html (hunk 1 above)
Comment (zcorpan) [reply uievents/textInput/smiley-manual.html:14]: Yes, as far as I know this isn't automatable in wpt.

**Review by masayuki-nakano — APPROVED** [review]
Thank you!

**Comment by zcorpan** [comment]
> The IDL should be imported from the spec after the spec change has landed.

I think that will be when https://github.com/w3c/webref/pull/1153 is merged and the corresponding automated PR for wpt is merged. (Example of such a PR: https://github.com/web-platform-tests/wpt/pull/44379 )

**Pull request merged**